### PR TITLE
[SPARK-45505][PYTHON] Refactor analyzeInPython to make it reusable

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

Currently, the `analyzeInPython` method in UserDefinedPythonTableFunction object can starts a Python process in driver and run a Python function in the Python process. This PR aims to refactor this logic into a reusable runner class.

### Why are the changes needed?

To make the code more reusable.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by allisonwang-db** [comment]
cc @ueshin

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    envVars.put("SPARK_LOCAL_DIRS", localdir)
+    if (reuseWorker) {
+      envVars.put("SPARK_REUSE_WORKER", "1")
+    }
+    if (simplifiedTraceback) {
+      envVars.put("SPARK_SIMPLIFIED_TRACEBACK", "1")
+    }
+    workerMemoryMb.foreach { memoryMb =>
+      envVars.put("PYSPARK_UDTF_ANALYZER_MEMORY_MB", memoryMb.toString)
+    }
```
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala:74]: This part seems to be UDTF analyzer specific so not reusable. should use a more generic name, or add another abstract method to additionally setup the `envVars`?

**Review by ueshin — COMMENTED** [review]
Otherwise, LGTM.
Thanks for the refactoring!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala (hunk 1 above)
Comment (allisonwang-db) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala:74]: Thanks for catching this. I think we can use a more generic name for this config and share it with all Python processes running on the driver side. We can have dedicated configs if needed in the future.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  protected def receiveFromPython(dataIn: DataInputStream): T
+
+  def runInPython(): T = {
+    val env = SparkEnv.get
+    val bufferSize: Int = env.conf.get(BUFFER_SIZE)
+    val authSocketTimeout = env.conf.get(PYTHON_AUTH_SOCKET_TIMEOUT)
+    val reuseWorker = env.conf.get(PYTHON_WORKER_REUSE)
+    val localdir = env.blockManager.diskBlockManager.localDirs.map(f => f.getPath()).mkString(",")
+    val simplifiedTraceback: Boolean = SQLConf.get.pysparkSimplifiedTraceback
+    val workerMemoryMb = SQLConf.get.pythonUDTFAnalyzerMemory
```
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala:54]: We also need to update the config name to be more generic?

Comment (allisonwang-db) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonPlannerRunner.scala]: Yup! Updated.

**Review by ueshin — APPROVED** [review]

**Comment by ueshin** [comment]
The failed tests seem not related to this PR. Let me merge this now.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**